libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/location/LocationViewModel.java

```diff
@@ -84,6 +84,8 @@ public void updateShouldSimulateRoute(boolean shouldSimulateRoute) {
    * @param route to be mocked
    */
   public void updateRoute(DirectionsRoute route) {
+    // MockLocationEngine is deactivated first to avoid weird behavior with subsequent navigation sessions
+    deactivateLocationEngine();
```

Comment:
Can we test with this removed now that we merged in the fix from https://github.com/mapbox/mapbox-navigation-android/pull/646?